### PR TITLE
Discard leaves and holidays

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
### What
- Discard leaves and leave types along with time off entries.
- Discard holidays and holiday info along with time off entries
- Updated the leave management calculations

### CODE REVIEW

File: app/models/holiday.rb [hunk 1]
```diff
@@ -16,18 +17,29 @@
...
+  include Discard::Model
+
   has_many :holiday_infos, dependent: :destroy
   belongs_to :company
 
   enum time_period_optional_holidays: { per_quarter: 0, per_year: 1, per_month: 2, per_week: 3 }
 
   validates :year, presence: true, uniqueness: { scope: :company_id }
   validates :year, numericality: { only_integer: true, greater_than_or_equal_to: 1900, less_than_or_equal_to: 2099 }
+
```
Comment (nisusam) [inline app/models/holiday.rb:37]: Add default_scope

```RUBY
default_scope -> { kept }
```

**Review by nisusam — APPROVED** [review]
LGTM!

Add `default_scope -> { kept }` in respective models.

File: app/models/holiday.rb (hunk 1 above)
Comment (prasanthchaduvula) [reply app/models/holiday.rb:37]: @vipulnsward Can we add default scopes?

Comment (vipulnsward) [reply app/models/holiday.rb:37]: We avoid default scope as much as possible.

**Pull request merged**